From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 20 Nov 2016 23:09:24 -0500
Date: Wed, 18 Aug 2021 21:08:20 +0200
Subject: dirmngr: Avoid automatically checking upstream swdb.

* dirmngr/dirmngr.c (housekeeping_thread): Avoid automatically
checking upstream's software database.  In Debian, software updates
should be handled by the distro mechanism, and additional upstream
checks only confuse the user.
* doc/dirmngr.texi: document that --allow-version-check does nothing.

Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
---
 dirmngr/dirmngr.c | 2 --
 doc/dirmngr.texi  | 7 ++++---
 2 files changed, 4 insertions(+), 5 deletions(-)

Patch-Source: https://sources.debian.org/data/main/g/gnupg2/2.2.27-2/debian/patches/dirmngr-idling/dirmngr-Avoid-automatically-checking-upstream-swdb.patch

diff --git a/dirmngr/dirmngr.c b/dirmngr/dirmngr.c
index 89eea4e..f26ed63 100644
--- a/dirmngr/dirmngr.c
+++ b/dirmngr/dirmngr.c
@@ -1955,8 +1955,6 @@ housekeeping_thread (void *arg)
   if (network_activity_seen)
     {
       network_activity_seen = 0;
-      if (opt.allow_version_check)
-        dirmngr_load_swdb (&ctrlbuf, 0);
       workqueue_run_global_tasks (&ctrlbuf, 1);
     }
   else
diff --git a/doc/dirmngr.texi b/doc/dirmngr.texi
index 843fdbf..84a8d28 100644
--- a/doc/dirmngr.texi
+++ b/doc/dirmngr.texi
@@ -291,9 +291,10 @@ Set the size of the queue for pending connections.  The default is 64.
 @item --allow-version-check
 @opindex allow-version-check
 Allow Dirmngr to connect to @code{https://versions.gnupg.org} to get
-the list of current software versions.  If this option is enabled
-the list is retrieved in case the local
-copy does not exist or is older than 5 to 7 days.  See the option
+the list of current software versions.
+On Alpine-packaged versions, this option does nothing since software
+updates should be handled by the distribution.
+See the option
 @option{--query-swdb} of the command @command{gpgconf} for more
 details.  Note, that regardless of this option a version check can
 always be triggered using this command:
